### PR TITLE
[NPU] Fix: recompute true-class CE gradient in fp32 on no-weight backward path

### PR DESCRIPTION
**Pull request by zheliuyu**
## Summary

PR #1329 added `test_correctness_true_class_grad_confident_predictions`, which compares the true-class logit gradient against an fp32 reference and requires Liger to be at least as accurate as PyTorch’s low-precision backward. On Ascend NPU, three cases fail when `label_smoothing=0.0` (`sum`/`mean`, bf16/fp16).

For plain cross-entropy (`no weight`, `no softcap`, `no z-loss`, `label_smoothing=0`), the Ascend backend uses `liger_cross_entropy_backward_kernel_no_weight`. That path wrote `softmax(x_y) * scale` into the low-precision `dX` buffer and then corrected the true-class entry with a read-modify-write (`load dX[y] - scale`). When `softmax(x_y) → 1`, rounding in bf16/fp16 before subtracting the `-1/N` term inflates error—the same failure mode fixed on GPU in #1329.

This change recomputes `dx_y = (softmax(x_y) - 1) * scale` once in fp32 from `X[y]` and `lse`, then overwrites `dX[y]`, matching the intent of the upstream Triton fix while keeping the existing O(V) loop unchanged.

## Performance

cross_entropy_speed_full_token_length

cross_entropy_memory_full_token_length

## Testing

```python
pytest test/transformers/test_cross_entropy.py 
```

image

- Hardware Type: All NPUs
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by zheliuyu** [comment]
@Tcc0403 This PR is ready for review. Thanks.

**Review by Tcc0403 — APPROVED** [review]
LGTM

**Comment by zheliuyu** [comment]
Hi @Tcc0403 — thanks again for reviewing #1341 and #1346. Both are approved and CI is green on our NPU runs. Could you help merge them when you have time? Happy to address any follow-ups if needed.
